src/renderer/components/+preferences/extensions.tsx

```diff
@@ -19,13 +19,21 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+import { withInjectables } from "@ogre-tools/injectable-react";
+import type { IComputedValue } from "mobx";
 import { observer } from "mobx-react";
 import React from "react";
-import { AppPreferenceRegistry } from "../../../extensions/registries";
+import type { AppPreferenceRegistration } from "./app-preferences/app-preference-registration";
+import appPreferencesInjectable from "./app-preferences/app-preferences.injectable";
 import { ExtensionSettings } from "./extension-settings";
 
-export const Extensions = observer(() => {
-  const settings = AppPreferenceRegistry.getInstance().getItems();
+interface Dependencies {
+  appPreferenceItems: IComputedValue<AppPreferenceRegistration[]>
+}
+
+const NonInjectedExtensions: React.FC<Dependencies> = ({ appPreferenceItems }) => {
```

Comment:
This needs to be `observer`. Which also means you can get rid of the type annotation

```
const NonInjectedExtensions = observer(({ appPreferenceItems }: Dependencies) => { ...
```

Reply:
Ah I see that you do that below...

Reply:
again, I took this pattern from #4615

Reply:
Fair, enough I don't think it matters that much.